### PR TITLE
FE-Listing - single listing

### PR DESCRIPTION
**Pull request by May-Tove**
Using the dummy API to be able to finish some of the tasks for now. I get an error on the fetch due to an issue in the apiBaseFetch.js file, this has been notified and should have been fixed in the main branch.

I have not made function to check if user is student or company just yet, but wanted to add edit and delete button now so we can work on the functions for this. I will also have to alter the template when design is finished.

The following tasks should be done in this branch:
- Create function to get single listing by id
- Replace hard coded HTML with dynamic JS template
- Create function to view single listing by id
- Add dynamic page title and meta description for single listing page

### CODE REVIEW

File: pages/listings/listing/index.html [hunk 1]
```diff
@@ -4,54 +4,22 @@
...
-            </div>
-          </div>
+        <div class="row d-flex alig-items-center justify-content-center my-5" id="singleListingContainer">
+          <!-- Content rendered in JS -->
         </div>
       </div>
     </main>
     <footer></footer>
     <script src="/index.js" type="module"></script>
+    <script src="/src/js/index.js" type="module"></script>
```
Comment (Jenfer93) [inline pages/listings/listing/index.html:22]: Is this and the line above the same?

**Review by Jenfer93 — COMMENTED** [review]
Other than one comment, this looks good to me! Åke will have to have a look to, and then it can be merged

File: pages/listings/listing/index.html (hunk 1 above)
Comment (Jenfer93) [reply pages/listings/listing/index.html:22]: ah! yeah, okei, I'm unsure too, lets see what Åke says to this :)

File: src/js/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { viewSingleListing } from './views/listings/index.js';
+
```
Comment (Christonn93) [inline src/js/index.js:2]: This one you can do in the main index.js file.

Comment (Jenfer93) [reply src/js/index.js:2]: Thank you Chris!

File: pages/listings/listing/index.html (hunk 1 above)
Comment (Christonn93) [reply pages/listings/listing/index.html:22]: dont need that last script. 
Add it in the root index. -
like this: 
image


then to run it you will add it in the correct route case like this: 

```js
import { viewSingleListing } from './views/listings/index.js';
import * as ui from './src/js/ui/index.js';

ui.displayBaseLayout();



/**
 * 
 * Just a router switch do change page title and run the needed functions for each page.
 * The router switch is requiring a id on the body element of the html page. 
 * 
 * @example // Profile UI settings
                case 'dashboard':
                // Page title
                document.querySelector('title').innerText = defaultTitle + ` || ` + `Dashboard`;
                break;
 * 
 * 
 */
const routerSwitch = () => {
  // Default/Fallback page title
  const defaultTitle = 'Add default title';

  const routeName = document.body.id;
  switch (routeName) {
    // Homepage UI settings
    case 'homepage':
      // Page title
      document.querySelector('title').innerText = defaultTitle + ` || Homepage`;
      break;

    case '{{ ADD YOU BODY ID HERE }}':
      // Page title
      document.querySelector('title').innerText = defaultTitle + ` || Homepage`;
      viewSingleListing()
      break;

    // 404 UI settings
    case '404':
      // Page title
      document.querySelector('title').innerText = defaultTitle + ` || ` + '404';
      break;
    default:
  }
};
routerSwitch();

**Review by akeek — APPROVED** [review]
This looks excellent! For the sake of progress, I will approve now without writing tests for every function. But it looks great!

**Pull request merged**